homeassistant/components/anglian_water/sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Anglian Water sensor platform."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from enum import StrEnum
+
+from pyanglianwater.meter import SmartMeter
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import AnglianWaterConfigEntry, AnglianWaterUpdateCoordinator
+from .entity import AnglianWaterEntity
+
+PARALLEL_UPDATES = 0
+
+
+class AnglianWaterSensor(StrEnum):
+    """Store keys for Anglian Water sensors."""
+
+    YESTERDAY_CONSUMPTION = "yesterday_consumption"
+    YESTERDAY_WATER_COST = "yesterday_water_cost"
+    YESTERDAY_SEWERAGE_COST = "yesterday_sewerage_cost"
+    LATEST_READING = "latest_reading"
+
+
+@dataclass(frozen=True, kw_only=True)
+class AnglianWaterSensorEntityDescription(SensorEntityDescription):
+    """Describes AnglianWater sensor entity."""
+
+    value_fn: Callable[[SmartMeter], float]
+
+
+ENTITY_DESCRIPTIONS: tuple[AnglianWaterSensorEntityDescription, ...] = (
+    AnglianWaterSensorEntityDescription(
+        key=AnglianWaterSensor.YESTERDAY_CONSUMPTION,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        device_class=SensorDeviceClass.WATER,
+        value_fn=lambda entity: entity.get_yesterday_consumption,
+        state_class=SensorStateClass.TOTAL,
+        translation_key=AnglianWaterSensor.YESTERDAY_CONSUMPTION,
+    ),
+    AnglianWaterSensorEntityDescription(
+        key=AnglianWaterSensor.LATEST_READING,
+        native_unit_of_measurement=UnitOfVolume.CUBIC_METERS,
+        device_class=SensorDeviceClass.WATER,
+        value_fn=lambda entity: entity.latest_read,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        translation_key=AnglianWaterSensor.LATEST_READING,
+    ),
+    AnglianWaterSensorEntityDescription(
+        key=AnglianWaterSensor.YESTERDAY_WATER_COST,
+        native_unit_of_measurement="GBP",
+        device_class=SensorDeviceClass.MONETARY,
+        value_fn=lambda entity: entity.yesterday_water_cost,
+        state_class=SensorStateClass.TOTAL,
+        translation_key=AnglianWaterSensor.YESTERDAY_WATER_COST,
+    ),
+    AnglianWaterSensorEntityDescription(
+        key=AnglianWaterSensor.YESTERDAY_SEWERAGE_COST,
+        native_unit_of_measurement="GBP",
+        device_class=SensorDeviceClass.MONETARY,
+        value_fn=lambda entity: entity.yesterday_sewerage_cost,
+        state_class=SensorStateClass.TOTAL,
+        translation_key=AnglianWaterSensor.YESTERDAY_SEWERAGE_COST,
+    ),
```

Comment:
I would argue that a device class is not fitting here

Reply:
May I ask why?

The monetary device class would correctly display the costs in the UI, whereas without it will just show "GBP" at the end which is incorrect for the UK:

<img width="320" height="312" alt="image" src="https://github.com/user-attachments/assets/ebe43af2-1d75-4e36-91bb-31d88ab5110d" />

(Note: this screenshot is from the custom integration which hasn't been updated for a while)

Reply:
You can't have measurement with monetary, as the unit of measurement is always wrong in that case. This is a discussion that has been done several times, I can link you to those, but should we remove the state classes for now so we can merge this and add it later?